Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py

```diff
@@ -3447,8 +3448,9 @@ def args_to_request_filters(args):
 def add_tag_to_endpoints_command(client: CoreClient, args: Dict):
     endpoint_ids = argToList(args.get('endpoint_ids', []))
     tag = args.get('tag')
-
-    raw_response = client.add_tag_endpoint(endpoint_ids=endpoint_ids, tag=tag, args=args)
+    raw_response = {}
+    for b in batch(endpoint_ids, 1000):
+        raw_response.update(client.add_tag_endpoint(endpoint_ids=b, tag=tag, args=args))
```

Comment:
I understand you might want to use these commands on a large number of endpoints. can you explain the general scenario?

Reply:
We use tags to manage the agent upgrade processes.  So we have a playbook that takes all the endpoint hostnames, gets the endpoint IDs, and then applies the tag.  We learned several months ago with the XDR devs that there is undocumented API limitations on the number of endpoint ids that can be in a single api call, hence the batching.